acme/resource_acme_certificate.go

```diff
@@ -130,10 +194,24 @@ func resourceACMECertificateUpdate(d *schema.ResourceData, meta interface{}) err
 	}
 
 	cert := expandCertificateResource(d)
-	if err := setDNSChallenge(client, d.Get("dns_challenge").([]interface{})[0].(map[string]interface{})); err != nil {
+
+	provider, err := NewDNSProviderWrapper()
+	if err != nil {
 		return err
 	}
 
+	var challenge_opts []dns01.ChallengeOption
+	for _, v := range d.Get("dns_challenge").([]interface{}) {
+		if p, opts, err := setDNSChallenge(client, v.(map[string]interface{})); err == nil {
+			provider.providers = append(provider.providers, p)
+			challenge_opts = append(challenge_opts, opts...)
+		} else {
+			return err
+		}
+	}
+
+	client.Challenge.SetDNS01Provider(provider)
```

Comment:
This should be `client.Challenge.SetDNS01Provider(provider, challenge_opts...)` so that recursive nameservers get set.

Reply:
PS: This also would need to be done for `Create`, but check out my comment [here](https://github.com/terraform-providers/terraform-provider-acme/pull/49#issuecomment-493250474) which probably supersedes this.